lib/rspec/core/bisect/example_minimizer.rb

```diff
@@ -18,20 +16,62 @@ def initialize(runner, reporter)
         def find_minimal_repro
           prep
 
-          self.remaining_ids = non_failing_example_ids
+          _, duration = track_duration do
+            bisect_over(non_failing_example_ids)
+          end
+
+          notify(:bisect_complete, :duration => duration,
+                                   :original_non_failing_count => non_failing_example_ids.size,
+                                   :remaining_count => remaining_ids.size)
+
+          remaining_ids + failed_example_ids
+        end
+
+        def bisect_over(candidate_ids)
+          notify(:bisect_dependency_check_started)
+          if get_expected_failures_for?([])
+            notify(:bisect_dependency_check_failed)
+            self.eliminated_ids = non_failing_example_ids
+            return
+          end
+          notify(:bisect_dependency_check_passed)
+
+          bisect_over_recursive(candidate_ids)
+        end
+
+        def bisect_over_recursive(candidate_ids)
```

Comment:
Instead of `bisect_over(candidate_ids)` and `bisect_over_recursive(candidate_ids)`, WDYT about calling the methods `bisect()` and `bisect_over(candidate_ids)`?  The former method is called only once and already assumes the `candidate_ids` list is `non_failing_example_ids` so it's not really needed as an argument.  Plus it performs the entire bisect procedure over the entire id set, not just over a subset, so `bisect` seems like a better method name.  I also prefer not to put `recursive` in a method name -- that's obvious from seeing it call itself.